cmd/modern/root/config/connection-strings.go

```diff
@@ -43,36 +45,88 @@ func (c *ConnectionStrings) run() {
 
 	// connectionStringFormats borrowed from "portal.azure.com" "connection strings" pane
 	var connectionStringFormats = map[string]string{
-		"ADO.NET": "Server=tcp:%s,%d;Initial Catalog=%s;Persist Security options=False;User ID=%s;Password=%s;MultipleActiveResultSets=False;Encode=True;TrustServerCertificate=False;Connection Timeout=30;",
-		"JDBC":    "jdbc:sqlserver://%s:%d;database=%s;user=%s;password=%s;encrypt=true;trustServerCertificate=false;loginTimeout=30;",
-		"ODBC":    "Driver={ODBC Driver 13 for SQL Server};Server=tcp:%s,%d;Database=%s;Uid=%s;Pwd=%s;Encode=yes;TrustServerCertificate=no;Connection Timeout=30;",
+		"ADO.NET": "Server=tcp:%s,%d;Initial Catalog=%s;Persist Security Info=False;User ID=%s;Password=%s;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=%s;Connection Timeout=30;",
+		"JDBC":    "jdbc:sqlserver://%s:%d;database=%s;user=%s;password=%s;encrypt=true;trustServerCertificate=%s;loginTimeout=30;",
+		"ODBC":    "Driver={ODBC Driver 18 for SQL Server};Server=tcp:%s,%d;Database=%s;Uid=%s;Pwd=%s;Encrypt=yes;TrustServerCertificate=%s;Connection Timeout=30;",
+		"GO":      "sqlserver://%s:%s@%s,%d?database=master;encrypt=true;trustServerCertificate=%s;loginTimeout=30",
+		"SQLCMD":  "sqlcmd -S %s,%d -U %s",
 	}
 
 	endpoint, user := config.CurrentContext()
+
 	if user != nil {
 		for k, v := range connectionStringFormats {
-			connectionStringFormats[k] = fmt.Sprintf(v,
-				endpoint.EndpointDetails.Address,
-				endpoint.EndpointDetails.Port,
-				"master",
-				user.BasicAuth.Username,
-				secret.Decode(user.BasicAuth.Password, user.BasicAuth.PasswordEncrypted))
-		}
+			if k == "GO" {
+				connectionStringFormats[k] = fmt.Sprintf(
+					"sqlserver://%s:%s@%s,%d?database=master;encrypt=true;trustServerCertificate=%s;loginTimeout=30",
```

Comment:
>"sqlserver://%s:%s@%s,%d?database=master;encrypt=true;trustServerCertificate=%s;loginTimeout=30", [](http://example.com/codeflow?start=5&length=97)

is this the same as line 51?
I think it'll be more sustainable to build a Uri using a Uri package instead of manually building a Uri with Sprintf. That way you ensure proper escaping etc.

Reply:
Fixed the dupe line

Reply:
Added issue:

[https://github.com/microsoft/go-sqlcmd/issues/265](https://github.com/microsoft/go-sqlcmd/issues/265)

for reimplementing using a Uri package